bdemeta/resolver.py

```diff
@@ -156,23 +157,36 @@ def _is_cmake(root: Path, name: str) -> Optional[Path]:
             return path
         return None
 
-    def identify(self, name: str) -> Identification:
+    def identify_root(self, name: str) -> Optional[Tuple[Path, Identification]]:
         for root in self._roots:
             path = TargetResolver._is_group(root, name)
             if path is not None:
-                return Identification('group', path)
+                return root, Identification('group', path)
 
             path = self._is_standalone(root, name)
             if path is not None:
-                return Identification('package', path)
+                return root, Identification('package', path)
 
             path = self._is_application(root, name)
             if path is not None:
-                return Identification('application', path)
+                return root, Identification('application', path)
 
             path = TargetResolver._is_cmake(root, name)
             if path is not None:
-                return Identification('cmake', path)
+                return root, Identification('cmake', path)
+
+        return None
+
+    def identify(self, name: str) -> Identification:
+        root_identity = self.identify_root(name)
+        if root_identity:
+            root, identification = root_identity
+
+            if root in self._conan_roots:
+                # This target will be provided by conan.
+                return Identification('conan', root)
```

Comment:
If we supported an explicit `conan_targets` property, this implementation could go back to its original simpler kind and just return `Identification('conan', path)`. It also looks like the `root` here is unused, right?

Reply:
`root` is used to select correct target, see line 267.